### PR TITLE
Chore/lint_dep_update

### PR DESCRIPTION
**Pull request by toniocodo**
- upgrade dependencies to fix security issues
- upgrade vite, remove deprecated plugins, update SSr config
- upgrade react-router to v6
- add eslint, husky and lint-staged for reformat on commit
- fix eslint on all codebase for consistency

### CODE REVIEW

**Review by rafaelugolini — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -25,51 +41,54 @@
     "@types/recharts": "^1.8.19",
     "@types/styled-components": "^5.1.7",
     "@types/three": "^0.128.0",
+    "@vitejs/plugin-react": "^1.3.2",
```
Comment (doncesarts) [inline package.json:44]: 👀  some tooling

Comment (toniocodo) [reply package.json:44]: yes, it's replacement for previous plugin

File: package.json [hunk 2]
```diff
@@ -25,51 +41,54 @@
...
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
     "react-helmet-async": "^1.0.9",
     "react-hook-inview": "^4.3.8",
     "react-loading-skeleton": "^2.2.0",
-    "react-resize-detector": "2.3.0",
-    "react-router-config": "^5.1.1",
-    "react-router-dom": "^5.2.0",
+    "react-resize-detector": "2.3.0",    
+    "react-router-dom": "6",
```
[the remark this answers is not in the document]
Comment (toniocodo) [reply package.json:56]: 👍 nice catch, this is leftover from my experimentation with router... I'll create new PR to pass it to latest version

**Comment by toniocodo** [comment]
@doncesarts update dep PR #86

**Pull request merged**